### PR TITLE
[#4883] Remove unused actions whose `@name` name starts with "__"

### LINKED ISSUE #4883: Unexpected "Duplicates declaration" error when P4 program contains no duplicate declarations
```
#include <core.p4>

struct S {
    bit<16> f;
}

extern bool foo(in bool x, in bool y);

control C(inout S s) {
    action a1() {
        if (foo(s.f == 0, false)) {
            return;
        }
    }

    @name("._xyz")
    action a2() {
        a1();
        a1();
    }

    table t {
        actions = { a2; }
    }

    apply {
        t.apply();
    }
}

control C2() {
    apply {
        S s;
        C.apply(s);
    }
}

control proto();
package top(proto p);

top(C2()) main;
```
results in:
```
[--Werror=duplicate] error: C.hasReturned: Duplicates declaration C.hasReturned
```

### PR DESCRIPTION
**Pull request by kfcripps**
Fixes #4883

### CODE REVIEW

File: frontends/p4/unusedDeclarations.cpp [hunk 1]
```diff
@@ -112,7 +112,7 @@ const IR::Node *RemoveUnusedDeclarations::process(const IR::IDeclaration *decl)
     LOG3("Visiting " << decl);
     if (decl->getName().name == IR::ParserState::verify && getParent<IR::P4Program>())
         return decl->getNode();
-    if (decl->getName().name.startsWith("__"))
+    if (decl->externalName(decl->getName().name).startsWith("__"))
```
Comment (fruffy) [inline frontends/p4/unusedDeclarations.cpp:115]: What happens when you have something like `@name("a2") action _xyz()` now?

Comment (kfcripps) [reply frontends/p4/unusedDeclarations.cpp:115]: @fruffy Please see the added test outputs

Comment (kfcripps) [reply frontends/p4/unusedDeclarations.cpp:115]: tagging @oleg-ran-amd for any additional comments

File: testdata/p4_16_samples_outputs/issue4883_dup_has_returned_decl2.p4-stderr [hunk 2]
```diff
@@ -0,0 +1,6 @@
+issue4883_dup_has_returned_decl2.p4(34): [--Wwarn=uninitialized_use] warning: s.f may be uninitialized
```
Comment (fruffy) [inline testdata/p4_16_samples_outputs/issue4883_dup_has_returned_decl2.p4-stderr:1]: Nit: This seems like noise which can be fixed by initializing s, making it part of C2's parameters. Feel free to ignore this if required for the test.

**Review by fruffy — APPROVED** [review]

**Comment by kfcripps** [comment]
> Approving this, but maybe clearing up the renaming behavior of inlining is the better long-term solution. Personally, I do not see why inlining should use the control-plane name.

@oleg-ran-amd Do you want to try this approach instead? If so, I will not merge this yet and allow you to push such changes to this branch (or open a new PR if you prefer that).

File: testdata/p4_16_samples_outputs/issue4883_dup_has_returned_decl2.p4-stderr (hunk 2 above)
Comment (kfcripps) [reply testdata/p4_16_samples_outputs/issue4883_dup_has_returned_decl2.p4-stderr:1]: I do not remember if it is required for the test. @oleg-ran-amd If you think it's not required, feel free to push such changes to this branch.

**Comment by kfcripps** [comment]
> > Approving this, but maybe clearing up the renaming behavior of inlining is the better long-term solution. Personally, I do not see why inlining should use the control-plane name.
> 
> @oleg-ran-amd Do you want to try this approach instead? If so, I will not merge this yet and allow you to push such changes to this branch (or open a new PR if you prefer that).

Or I can just merge this now and let you make the better long-term changes later. Let me know what you prefer.

**Comment by kfcripps** [comment]
@oleg-ran-amd From @asl:
> for 4900 – I’d rather implement proper approach with renaming rather than doing band-aid later

**Comment by kfcripps** [comment]
Per DM with @oleg-ran-amd, I will merge this now and leave #4883 open for Oleg to look into the suggested better long-term solution later.

**Pull request merged**